### PR TITLE
Improve fork config generation for berkeley

### PR DESCRIPTION
**Pull request by Sventimir**
Explain your changes:
* Merge changes from #14266 before they can land in `compatible`.

Explain how you tested your changes:
* Connect to a testnet and extract the ledger.
* CI.

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them

* Closes https://github.com/MinaFoundation/mina/issues/95

### CODE REVIEW

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by snaitmouloud-mf** [comment]
@bkase any chance to review this PR ?

**Comment by Sventimir** [comment]
!ci-build-me

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
Can you cherry-pick the changes you did here: https://github.com/MinaProtocol/mina/pull/14285/commits/27157c9f2cfccea40304fcdd826197888936ee5d into this PR too?

**Comment by ejMina226** [comment]
Hi @ghost-not-in-the-shell . I've cherry-picked the changes. Can you review when you are free and I can merge? Thanks.

**Comment by ejMina226** [comment]
!ci-build-me

File: src/lib/runtime_config/runtime_config.ml [hunk 1]
```diff
@@ -1,9 +1,22 @@
 open Core_kernel
 
 module Fork_config = struct
+  (* Note that previous_length might be smaller than the gernesis_slot
+     or equal if a block was produced in every slot possible. *)
   type t =
-    { previous_state_hash : string; previous_length : int; genesis_slot : int }
-  [@@deriving yojson, dhall_type, bin_io_unversioned]
```
Comment (ejMina226) [inline src/lib/runtime_config/runtime_config.ml:6]: @ghost-not-in-the-shell Whilst cherry picking I missed out the dhall_type. Does this need to be added in? Thanks.

Comment (ghost-not-in-the-shell) [reply src/lib/runtime_config/runtime_config.ml:6]: I think dhall_type is not implemented on develop branch, so you can safely change it to `[@@deriving yojson, bin_io_unversioned]`.

Comment (ejMina226) [reply src/lib/runtime_config/runtime_config.ml:6]: Thanks. I'll leave it as is then (with the dhall_type omitted).

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by ejMina226** [comment]
!ci-build-me

**Review by MartinMinkov — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by nholland94** [comment]
Heads up: I've reverted https://github.com/MinaProtocol/mina/pull/14285 for now, as this PR needs reviewed first. Any suggested changes on that merge PR should be ported back to here, and the merge PR can be recreated once the review here is completed (if still necessary).

**Comment by Sventimir** [comment]
This will be redone after https://github.com/MinaProtocol/mina/pull/14266 gets approved to more easily include all the commits that went into it.

**Pull request closed without merging**